src/db/sysdb_search.c

```diff
@@ -25,6 +25,9 @@
 #include <time.h>
 #include <ctype.h>
 
+#define IS_SYSDB_NAME(a) (strcmp((a), SYSDB_NAME) == 0)
```

Comment:
If you think that this helper is really needed, then imo it would be better to use `static inline` function with a check that `a != NULL` before calling `strcmp()`.

Reply:
In a normal situation I would agree with you, but let me disagree this time.

The check for `a != NULL` would be redundant because it is already checked. We could remove those checks and let this inline function make the safety check, but in this particular case this is more than just a safety check. Whether `attr` is `NULL` or not, has its own meaning. `attr` is `NULL` when `ListByName()` was called; and is not `NULL` for `ListByAttr()`. Because of this, I'd like to make this check evident and not hide it inside a function.

In addition, this macro is used only twice, and one of the checks is not even done in the same `if` statement (sysdb_search.c:552), changing the meaning is we remove it and making it redundant if we don't.

Reply:
But you never know how this helper will be used in the future.
Even if check would be redundant in current usage, it won't hurt because compiler will optimize it away (for inline function for sure).
If you don't mean this helper to be used anywhere else, I probably better wouldn't introduce it.

Reply:
OK